### ISSUE TITLE
Documentation Errors and Should We Wait Isar?

### ISSUE THREAD

**Issue by Peng-Qian** [isar/isar#122]
Hi,

I have closely watch Hive(update plan)/Isar for long time, and it is real struggling to wait Isar, is it possible to give us a promising instruction that when can we use stable Isar?

Isar Documentation currently have many errors on Document website and dart pub

```dart
@IsarCollection() // I think should be collection
class Teacher {
    int id;  // int?

    String subject;
}

@IsarCollection() // I think should be collection
class Student {
    int id; // int?

    String name; // should add late?

    final teachers = IsarLink<Teacher>(); // should be var since this will be re-assigned
}
```

**Comment by om-ha** [comment]
I think Isar is a very promising project. A hidden gem possibly the best in industry.

These are few mishaps here and there, in documentation and some bugs.

For example, I had to manually add a missing import in `isar.g.dart` (I wrote a bash script for that to make this more standardized and automated) But to be fair this missing import stems from a not-so-common but perfectly acceptable and cleaner approach for doing imports. Which is using a single importable file that exports multiple files. (see [this](https://stackoverflow.com/questions/55579092/how-to-avoid-writing-an-import-for-every-single-file-in-dart-flutter), [this](https://stackoverflow.com/questions/60796310/flutter-is-there-an-effect-on-having-a-single-import-rather-than-multiple), and [this](https://betterprogramming.pub/how-to-export-multiple-dart-files-from-just-one-b2f2964029e?gi=c7009c7b31c4))



With these corrected, I think Isar would be hugely adopted even more than Hive. But for myself I'm adopting Isar anyways! Because it has a huge potential.
